### PR TITLE
[ext-docs] Restructure Permission warnings

### PR DESCRIPTION
**Pull request by AmySteam**
## Description

This pull request aims to restructure the permissions warnings article in order to improve clarity and provide more explicit explanations of the most important behaviors related to permission warnings in Chrome extensions. Additionally, the history warning message will be updated to reflect the following: "Read and change your browsing history on all your signed-in devices."

## Changes Made:

- Restructured the permissions warnings article to enhance clarity and improve understanding.
- Add instruction on how to test extension updates when introducing a new permission with warning.
- Updated the history warning message to accurately describe the permission's behavior.
- Tweaks Permissions article.
- Fixes links to Permissions API.

## Staged links

- https://pr-6372-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/mv3/declare_permissions/
- https://pr-6372-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/mv3/manifest/activeTab/
- https://pr-6372-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/mv3/permission_warnings/
- https://pr-6372-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/reference/permissions/

## TODOs

- [ ] Waiting on new tool that makes it easier to test permission warnings @oliverdunk

### CODE REVIEW

**Review by oliverdunk — CHANGES_REQUESTED** [review]
Left one comment, overall this is looking really good and the instructions for the update testing tool seem perfect!

I'm curious how you feel about the overall structure of these pages. It feels like right now we have three different pages that are all some mix of best practices, explanation of what permissions are and advice on how to test things.

A few ideas:

- The vast majority of the content on https://pr-6372-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/reference/permissions/ could move to a new "How to use optional permissions" tutorial page (it's so much to scroll through to get to the API docs...)
- The best practices from https://pr-6372-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/mv3/permission_warnings/#permissions_with_warnings could mostly live alongside general best practices
- The "How to test permissions warnings" from https://pr-6372-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/mv3/permission_warnings/#permissions_with_warnings could go on a new "How to test permission warnings" page
- The "Permission warnings list" from https://pr-6372-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/mv3/permission_warnings/#permissions_with_warnings could go on a new "List of permission warnings" page
- The steps to grant files access could go on https://pr-6372-static-dot-dcc-staging.uc.r.appspot.com/docs/extensions/mv3/declare_permissions/ and we could rename that page to like "Requesting permissions" rather than "Declaring permissions"

Open to your thoughts and definitely get if we want to save this for another PR, but it just feels like to find say (the update instructions) right now you basically have to click through a bunch of long pages with overlapping names to try and find the one you want.

File: site/en/docs/extensions/mv3/permission_warnings/index.md [hunk 1]
```diff
@@ -1,155 +1,265 @@
...
-The [optional permissions sample extension's][gh-opt-perms] core functionality is overriding the new tab page.
-One feature is displaying the user's goal of the day. This feature only requires the [storage][api-storage]
-permission, which does not include a warning.
+This permission does **_not_** display a permission warning. It grants temporary host permission to
+the site the user is on. For more details, see [Understanding the activeTab
+permission][doc-activetab].
 
-{% Img src="image/BrQidfK9jaQyIHwdw91aVpkPiib2/wtbjayBDYDyKZe2x580P.png",
-       alt="Extension button that enables additional features", height="350", width="395" %}
+### Onboard users to allow access {: #allow_access }
```
[the remark this answers is not in the document]
Comment (AmySteam) [reply site/en/docs/extensions/mv3/permission_warnings/index.md:57]: Yeah I struggled to find a place. I'll probably end up moving this to declare permissions. Do you mind if I do this on a separate PR?

File: site/en/docs/extensions/reference/permissions/index.md [hunk 2]
```diff
@@ -2,7 +2,21 @@
...
+## Overview {: #overview }
+
+Permission warnings exist to describe the capabilities granted by an API, but some of these warnings may not be obvious. The Permissions API allows developers to explain permission warnings and introduce new features gradually which gives users a risk-free introduction to the extension. This way, users can specify how much access they are willing to grant and which features they want to enable.
+
+For example, the [optional permissions extension's][gh-optional-perms] core functionality is overriding the new tab page. One feature is displaying the user's goal of the day. This feature only requires the [storage][api-storage] permission, which does not include a warning. The extension has an additional feature, that users can enable by clicking the following button: 
+
+{% Img src="image/BrQidfK9jaQyIHwdw91aVpkPiib2/wtbjayBDYDyKZe2x580P.png",
+       alt="Extension button that enables additional features", height="350", width="395" %}
+
+Displaying the user's top sites requires the [topSites][api-top-sites] permission, which has the following warning.
```
[the remark this answers is not in the document]
Comment (AmySteam) [reply site/en/docs/extensions/reference/permissions/index.md:14]: Fixed it 👍

File: site/en/docs/extensions/mv3/permission_warnings/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (oliverdunk) [reply site/en/docs/extensions/mv3/permission_warnings/index.md:57]: Sure 👍

**Comment by AmySteam** [comment]
> I'm curious how you feel about the overall structure of these pages. It feels like right now we have three different pages that are all some mix of best practices, explanation of what permissions are and advice on how to test things.

I want to work on these more for sure. This was most certainly an MVP. I had other ideas for future iterations 😄 

> A few ideas:

I'll create separate tickets for these :)

> Open to your thoughts and definitely get if we want to save this for another PR, but it just feels like to find say (the update instructions) right now you basically have to click through a bunch of long pages with overlapping names to try and find the one you want.

I'll work on this a bit more today to decrease the scrolling bit. But I do think viewing and updating have to be displayed on separate sections.

**Review by oliverdunk — APPROVED** [review]
Thanks so much for all of the work on this!

**Review by IanStanion-google — APPROVED** [review]
LGTM! Nice job!

**Pull request merged**